pkg/ipcache/metadata.go

```diff
@@ -366,14 +372,12 @@ func (ipc *IPCache) doInjectLabels(ctx context.Context, modifiedPrefixes []cmtyp
 		unmanagedPrefixes = make(map[cmtypes.PrefixCluster]Identity)
 	)
 
-	ipc.metadata.RLock()
```

Comment:
This is not safe: `resolveIdentity` iterates over `ipc.metadata.m` (via `mergeParentLabels`) which now can be concurrently modified, which can cause a panic